### PR TITLE
Ability to render installer.ps1.

### PR DESCRIPTION
**Pull request by sersut**
This PR goes with https://github.com/chef/omnitruck/pull/158 and adds the ability to render `install.ps1` in omnitruck. 

Here is the diff that is generated with mixlib-install and one taken from omnitruck. Only one function is relocated in the file.

```
--- install_original.ps1	2015-12-08 14:40:34.000000000 -0800
+++ install_mixlib.ps1	2015-12-08 14:45:29.000000000 -0800
@@ -31,6 +31,44 @@
   }
 }
 
+function Test-ProjectPackage {
+  [cmdletbinding()]
+  param ($Path, $Algorithm = 'SHA256', $Hash)
+
+  if (-not (get-command get-filehash))   {
+    function disposable($o){($o -is [IDisposable]) -and (($o | get-member | foreach-object {$_.name}) -contains 'Dispose')}
+    function use($obj, [scriptblock]$sb){try {& $sb} catch [exception]{throw $_} finally {if (disposable $obj) {$obj.Dispose()}} }
+    function Get-FileHash ($Path, $Algorithm) {
+      $Path = (resolve-path $path).providerpath
+      $hash = @{Algorithm = $Algorithm; Path = $Path}
+      if ($Algorithm -like 'MD5') {
+        use ($c = New-Object -TypeName Security.Cryptography.MD5CryptoServiceProvider) {
+          use ($in = (gi $path).OpenRead()) {
+            $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
+          }
+        }
+      }
+      elseif ($Algorithm -like 'SHA256') {
+        use ($c = New-Object -TypeName Security.Cryptography.SHA256CryptoServiceProvider) {
+          use ($in = (gi $path).OpenRead()) {
+            $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
+          }
+        }
+      }
+      new-object PSObject -Property $hash
+    }
+  }
+  Write-Verbose "Testing the $Algorithm hash for $path."
+  $ActualHash = (Get-FileHash -Algorithm $Algorithm -Path $Path).Hash.ToLower()
+  Write-Verbose "`tDesired Hash - '$hash'"
+  Write-Verbose "`tActual Hash  - '$ActualHash'"
+  $Valid = $ActualHash -eq $Hash
+  if (-not $Valid) {
+    Write-Error "Failed to validate the downloaded installer.  The expected $Algorithm hash was '$Hash' and the actual hash was '$ActualHash' for $path"
+  }
+  return $Valid
+}
+
 function Get-ProjectMetadata {
   <#
     .SYNOPSIS
@@ -119,44 +157,6 @@
   $package_metadata
 }
 
-function Test-ProjectPackage {
-  [cmdletbinding()]
-  param ($Path, $Algorithm = 'SHA256', $Hash)
-
-  if (-not (get-command get-filehash))   {
-    function disposable($o){($o -is [IDisposable]) -and (($o | get-member | foreach-object {$_.name}) -contains 'Dispose')}
-    function use($obj, [scriptblock]$sb){try {& $sb} catch [exception]{throw $_} finally {if (disposable $obj) {$obj.Dispose()}} }
-    function Get-FileHash ($Path, $Algorithm) {
-      $Path = (resolve-path $path).providerpath
-      $hash = @{Algorithm = $Algorithm; Path = $Path}
-      if ($Algorithm -like 'MD5') {
-        use ($c = New-Object -TypeName Security.Cryptography.MD5CryptoServiceProvider) {
-          use ($in = (gi $path).OpenRead()) {
-            $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
-          }
-        }
-      }
-      elseif ($Algorithm -like 'SHA256') {
-        use ($c = New-Object -TypeName Security.Cryptography.SHA256CryptoServiceProvider) {
-          use ($in = (gi $path).OpenRead()) {
-            $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
-          }
-        }
-      }
-      new-object PSObject -Property $hash
-    }
-  }
-  Write-Verbose "Testing the $Algorithm hash for $path."
-  $ActualHash = (Get-FileHash -Algorithm $Algorithm -Path $Path).Hash.ToLower()
-  Write-Verbose "`tDesired Hash - '$hash'"
-  Write-Verbose "`tActual Hash  - '$ActualHash'"
-  $Valid = $ActualHash -eq $Hash
-  if (-not $Valid) {
-    Write-Error "Failed to validate the downloaded installer.  The expected $Algorithm hash was '$Hash' and the actual hash was '$ActualHash' for $path"
-  }
-  return $Valid
-}
-
 function Install-Project {
   <#
     .SYNOPSIS
@@ -253,4 +253,5 @@
 }
 set-alias install -value Install-Project
 export-modulemember -function 'Install-Project','Get-ProjectMetadata' -alias 'install'
+
 }
```

### CODE REVIEW

**Comment by sersut** [comment]
Also @jaym might see something that we don't in this :smile:

File: lib/mixlib/install/generator/powershell/scripts/get_project_metadata.ps1.erb [hunk 1]
```diff
@@ -36,9 +36,18 @@ function Get-ProjectMetadata {
     # Release channel to install from
     [validateset('current', 'stable')]
     [string]
-    $channel = 'stable'
+    $channel = 'stable',
+    # The following legacy switches are just aliases for the current channel
+    [switch]
+    $prerelease,
+    [switch]
+    $nightlies
```
Comment (jaym) [inline lib/mixlib/install/generator/powershell/scripts/get_project_metadata.ps1.erb]: are we allowed to use this word? I feel like @schisamo set up some sort of punishment for doing such.
If we never supported these for the powershell script, we can probably dump them.

**Comment by jaym** [comment]
LGTM. I wouldn't carry around the legacy stuff if we can get away with it.

**Comment by sersut** [comment]
@jaym removing the legacy stuff need to be left to a different PR I think. Otherwise when migrating to new omnitruck we won't be able to support old $things. I think it is OK the backcompat logic is very contained.

File: lib/mixlib/install/generator/base.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#
+
+require "erb"
+require "ostruct"
+require "mixlib/install/backend/omnitruck"
+
+module Mixlib
+  class Install
+    class Generator
+      class Base
```
Comment (patrick-wright) [inline lib/mixlib/install/generator/base.rb]: :+1:

**Comment by schisamo** [comment]
:+1:

**Comment by patrick-wright** [comment]
:+1:

**Comment by jaym** [comment]
install.ps1 does not work in old omnitruck, so there is no backwards compat stuff there

**Pull request merged**